File: apina-core/src/test/java/fi/evident/apina/java/model/JavaFieldTest.java
package fi.evident.apina.java.model;

import fi.evident.apina.java.model.type.JavaBasicType;
import org.junit.Test;
import org.objectweb.asm.Opcodes;

import static org.hamcrest.CoreMatchers.is;
import static org.junit.Assert.assertThat;

public class JavaFieldTest {

    @Test
    public void modifiers() {
        assertThat(fieldWithModifiers(0).isStatic(), is(false));
        assertThat(fieldWithModifiers(0).isEnumConstant(), is(false));
        assertThat(fieldWithModifiers(Opcodes.ACC_STATIC).isStatic(), is(true));
        assertThat(fieldWithModifiers(Opcodes.ACC_ENUM).isEnumConstant(), is(true));
    }

    private static JavaField fieldWithModifiers(int modifiers) {
        return new JavaField("foo", JavaVisibility.PUBLIC, new JavaBasicType("java.lang.String"), modifiers);
    }
}

File: apina-core/src/test/kotlin/fi/evident/apina/java/model/JavaFieldTest.kt
package fi.evident.apina.java.model

import fi.evident.apina.java.model.type.JavaType
import org.junit.jupiter.api.Test
import org.objectweb.asm.Opcodes
import kotlin.test.assertFalse
import kotlin.test.assertTrue

class JavaFieldTest {

    @Test
    fun modifiers() {
        assertFalse(fieldWithModifiers(0).isStatic)
        assertFalse(fieldWithModifiers(0).isEnumConstant)
        assertTrue(fieldWithModifiers(Opcodes.ACC_STATIC).isStatic)
        assertTrue(fieldWithModifiers(Opcodes.ACC_ENUM).isEnumConstant)
        assertTrue(fieldWithModifiers(Opcodes.ACC_TRANSIENT).isTransient)
    }

    private fun fieldWithModifiers(modifiers: Int) =
        JavaField("foo", JavaVisibility.PUBLIC, JavaType.basic<String>(), modifiers)
}
